CONTRIBUTING.md: Ask folks to disclose AI-written PRs (#58666)
JuliaLang/julia: CONTRIBUTING.md

* Review discussions on the [Julia Discourse forum](https://discourse.julialang.org).

> * If your pull request was generated by a generative AI tool, please disclose that.

Reviewer: ```suggestion
* If your pull request contains substantial contributions from a generative AI tool, please disclose that.
```

> * If your pull request contains substantial contributions from a generative AI tool, please disclose that.

Reviewer 2: ```suggestion
* If your pull request contains substantial contributions from a generative AI tool, please disclose so with details, and review all changes before opening. 
```
